dashboard/test/models/script_level_test.rb

```diff
@@ -607,6 +607,85 @@ class ScriptLevelTest < ActiveSupport::TestCase
     assert_equal false, script_level.hidden_for_section?(section.id)
   end
 
+  def create_visible_after_script
+    level = create :maze, name: 'maze 1', level_num: 'custom'
+    script = create :script
+    stage = create :stage, name: 'stage 1', script: script, visible_after: '2020-04-01 08:00:00 -0700'
+    script_level = create :script_level, levels: [level], stage: stage, script: script
+
+    script_level
+  end
+
+  test 'valid_progression_level returns true for script level in stage with past visible after date for levelbuilder' do
+    Timecop.freeze(Time.new(2020, 4, 2))
+    levelbuilder = create :levelbuilder
+    script_level = create_visible_after_script
+
+    assert script_level.valid_progression_level?(levelbuilder)
```

Comment:
In this particular kind of test, you can save yourself some copying and pasting by doing all the setup and then running through multiple assertions. in this case, each of the other test cases you've got for 4/2 could just be a single additional line added here:
```
    assert script_level.valid_progression_level?(@levelbuilder)
    assert script_level.valid_progression_level?(@teacher)
    assert script_level.valid_progression_level?(@student)
```
For this to work, you need to have defined the users in the setup or setup_all blocks earlier.

I'm mentioning this to help you for next time so please don't feel obligated to update everything you've already got working here.